meson.build

```diff
@@ -213,7 +213,7 @@ if ucx_dep.found() and cuda_dep.found() and nvcc_prog.found()
     have_gpu_side = cuda.compiles('''
             #include <ucp/api/device/ucp_device_impl.h>
             int main() { return 0; }
-        ''', dependencies : ucx_dep, args: nvcc_flags)
+        ''', dependencies : [ucx_dep, doca_gpunetio_dep], args: nvcc_flags)
```

Comment:
let's add "TODO: expose gpunetIo dependency through ucx"
since for examaple, in theory ucx can be built without gpunetio and support only device api using cuda_ipc